lib/prawn/table/cell/text.rb

```diff
@@ -135,7 +135,13 @@ def text_box(extra_options={})
         #
         def styled_width_of(text)
           options = @text_options.reject { |k| k == :style }
-          with_font { @pdf.width_of(text, options) }
+          if text.empty?
```

Comment:
You are correct, `width_of` handles it correctly, but line 142 would subsequently call `#max` on an empty array and return nil; thus the conditional.